src/client/pythonEnvironments/discovery/locators/services/pipEnvHelper.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as path from 'path';
+import { getEnvironmentVariable } from '../../../../common/utils/platform';
+import { pathExists, readFile } from '../../../common/externalDependencies';
+
+/**
+ * Returns the path to Pipfile associated with the provided directory.
+ * @param searchDir the directory to look into
+ * @param lookIntoParentDirectories set to true if we should also search for Pipfile in parent directory
+ */
+async function getAssociatedPipfile(
+    searchDir: string,
+    lookIntoParentDirectories: boolean,
+): Promise<string | undefined> {

```

Comment:
Boolean args (in languages like TypeScript that don't support keyword arguments) aren't great for callers.  One solution is to use an object unpacking parameter:

```suggestion
async function getAssociatedPipfile(
    searchDir: string,
    { lookIntoParentDirectories: boolean },
): Promise<string | undefined> {
```

Reply:
FWIW, we could also make `depthToSearch` the parameter, where the default is 1 and a negative value means "look into parent directories".  However, that would be worth doing only if there were cases where we had different positive `depthToSearch` values, which we don't. So `lookIntoParentDirectories` has a better focus and consequently is better for readers.

Reply:
Right, I am not including `depthToSearch` as we don't have different values which needs searching.